### PR TITLE
chore: switch from CB to GHA for lint

### PR DESCRIPTION
**Pull request by bharathkkb**

### CODE REVIEW

**Review by Shabirmean — DISMISSED** [review]

**Comment by Shabirmean** [comment]
@bharathkkb I think the openstack directory is something I can fix. I just haven't had the time. Can you add `anthos-multi-cloud` to the exclude directories? I think that is the one which has a lot of linting issues that is not owned by the DPE team

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: 'lint'
+    runs-on: 'ubuntu-latest'
+    env:
+      EXCLUDE_LINT_DIRS: "\\./anthos-bm-openstack-terraform/resources"
```
Comment (NimJay) [inline .github/workflows/lint.yaml:34]: **Heads-up:**
This `EXCLUDE_LINT_DIRS` value doesn't seem to work.
We still see lint failures in [this lint build](https://github.com/GoogleCloudPlatform/anthos-samples/actions/runs/5126111306/jobs/9220175029?pr=612), in the `anthos-bm-openstack-terraform/resources` folder.

Comment (NimJay) [inline .github/workflows/lint.yaml:34]: **Question:**
Is there an environment variable that lets us ignore certain rules?
I was wondering if we could ignore [SC2181](https://www.shellcheck.net/wiki/SC2181), which seems overkill for long commands such as [this long command](https://github.com/GoogleCloudPlatform/anthos-samples/blob/main/anthos-bm-openstack-terraform/resources/abm_setup_gcp.sh#L24) which the linter is complaining about. Related issue: https://github.com/koalaman/shellcheck/issues/2465.

**Review by apeabody — APPROVED** [review]

**Review by apeabody — APPROVED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (apeabody) [reply .github/workflows/lint.yaml]: These can now be passed in: https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/terraform/test-org/org/locals.tf

**Pull request merged**